ios/Classes/GoogleMapsNavigationView.swift

```diff
@@ -49,52 +50,59 @@ class GoogleMapsNavigationView: NSObject, FlutterPlatformView, ViewSettledDelega
   var isAttachedToSession: Bool = false
 
   func view() -> UIView {
-    _navigationView
+    _mapView
   }
 
   init(frame: CGRect,
        viewIdentifier viewId: Int64,
+       isNavigationView: Bool,
        viewRegistry registry: GoogleMapsNavigationViewRegistry,
-       navigationViewEventApi: NavigationViewEventApi,
+       viewEventApi: ViewEventApi,
        navigationUIEnabledPreference: NavigationUIEnabledPreference,
        mapConfiguration: MapConfiguration, imageRegistry: ImageRegistry) {
     _viewId = viewId
+    _isNavigationView = isNavigationView
     _viewRegistry = registry
-    _navigationViewEventApi = navigationViewEventApi
+    _viewEventApi = viewEventApi
     _mapConfiguration = mapConfiguration
     _imageRegistry = imageRegistry
 
     let mapViewOptions = GMSMapViewOptions()
     _mapConfiguration.apply(to: mapViewOptions, withFrame: frame)
-    _navigationView = ViewStateAwareGMSMapView(options: mapViewOptions)
-    _mapConfiguration.apply(to: _navigationView)
+    _mapView = ViewStateAwareGMSMapView(options: mapViewOptions)
```

Comment:
I think we should use this `ViewStateAwareGMSMapView` wrapper for mapViews as well.
See `_mapView.viewSettledDelegate = self`
and `onViewSettled` method in this class.
The `ViewStateAwareGMSMapView` is wrapper that handles some of the issues while using mapviews inside flutter as flutter does not give proper size for the view before it is attached to the hierarhcy.
This wrapper allows to handle the view initialization (like camera) after the size is available.

Reply:
What do you mean by `for mapViews as well`? It is currently used for both map types. There is no separate class for navigation maps on iOS side.

Reply:
True, you are right!
I accidentally looked the Android implementation.